File: haskell/haskell-first-principles/ch04-basic-datatypes/tuple2.hs

fn :: (a, b) -> (c, d) -> ((b, d), (a, c))
fn tupA tupB =
  ((snd tupA, snd tupB), (fst tupA, fst tupB))

--
-- :load tuple2.hs
-- *Main> fn ("master", "yoda") ("jedi", "tux")
-- (("yoda","tux"),("master","jedi"))

File: old-notes/scala/effective-scala/week2/tuple_2.scala
package tuple_2

val point = (4, -3)
val yoda = ("Yoda", "The Force", 100)

//
// Can access by index (Haskell has fst, snd). Scala can access
// any tuple index.
//
val x = point(0)
val y = point(1)

val level = yoda(2)

@main def main(): Unit =
  println(x)
  println(y)
  println(level)
